crypto/des/des.c

```diff
@@ -349,9 +349,38 @@ void DES_set_key(const DES_cblock *key, DES_key_schedule *schedule) {
   }
 }
 
+// SP 800-67r2 section 2, the last bit of each byte in DES_cblock.bytes is used
+// for parity. The parity bits should be set to the complement of the modulo 2
+// sum of the previous seven bits
+static int DES_check_key_parity(const DES_cblock *key) {
+  uint8_t result = UINT8_MAX;
+
+  for (size_t i = 0; i < DES_KEY_SZ; i++) {
+    uint8_t b = key->bytes[i];
+    b ^= b >> 4;
+    b ^= b >> 2;
+    b ^= b >> 1;
+    result &= constant_time_eq_8(b & 1, 1);
+  }
+  return result & 1;
+}
+
+int DES_set_key(const DES_cblock *key, DES_key_schedule *schedule)
```

Comment:
Please add some documentation here and explain what the return value means and that you always set the key.